### PR TITLE
Transition to .NET Core CLI

### LINKED ISSUE #4280: Build with dotnet
This issues it to track the dotnet progress for EF.

cc @bricelam @divega

### PR DESCRIPTION
**Pull request by bricelam**
Changes
* Use KoreBuild-dotnet (resolves #4280)
* Remove **csproj**-based solution
* Remove netcore50 targets
* Combine net451 & dnx451
* Remove DNX commands (see #3925)

TODO
* Verify packages on UWP
* Verify CI builds (including xplat & perf)
* Fix tests that fail on xproj

### CODE REVIEW

**Comment by divega** [comment]
I noticed in tests (I think) we keep lots of references to DNX451 and DNXCORE50. Are those going away in the future?

**Comment by divega** [comment]
:shipit:

File: build.sh [hunk 1]
```diff
@@ -11,33 +24,25 @@ else
...
-if test ! -d packages/Sake; then
-    mono .nuget/nuget.exe install KoreBuild -ExcludeVersion -o packages -nocache -pre
-    mono .nuget/nuget.exe install Sake -ExcludeVersion -Source https://www.nuget.org/api/v2/ -Out packages
+    cp $cacheNuget $nugetPath
 fi
 
-if ! type dnvm > /dev/null 2>&1; then
-    source packages/KoreBuild/build/dnvm.sh
+if test ! -d $koreBuildFolder; then
+    mono $nugetPath install KoreBuild-dotnet -ExcludeVersion -o $buildFolder -nocache -pre
```
Comment (natemcmaster) [inline build.sh]: :cry: so close to actually getting rid of mono in our build

File: src/EntityFramework.Core/DbSet`.cs [hunk 2]
```diff
@@ -10,10 +10,10 @@
 using Microsoft.Data.Entity.ChangeTracking;
 using Microsoft.Data.Entity.Infrastructure;
 using Microsoft.Data.Entity.Query.Internal;
-#if NET451 || DNX451
+#if NET451
 using System.ComponentModel;
 #endif
-#if NET451 || DNX451
+#if NET451
```
Comment (natemcmaster) [inline src/EntityFramework.Core/DbSet`.cs]: Can merge into one #if section

File: src/EntityFramework.Core/project.json [hunk 3]
```diff
@@ -28,81 +24,60 @@
...
-        "System.Reflection.Extensions": "4.0.1-*",
-        "System.Reflection.TypeExtensions": "4.1.0-*",
-        "System.Resources.ResourceManager": "4.0.1-*",
-        "System.Runtime.Extensions": "4.0.11-*",
-        "System.Runtime.InteropServices": "4.0.21-*",
-        "System.Threading": "4.0.11-*",
-        "System.Threading.Tasks": "4.0.11-*"
-      }
-    },
-    "netcore50": {
```
Comment (natemcmaster) [inline src/EntityFramework.Core/project.json]: (Part of UWP verification you mentioned) Will the `build/netcore50/*.targets` still work?

File: test/EntityFramework.Commands.FunctionalTests/project.json [hunk 4]
```diff
@@ -4,29 +4,16 @@
...
     "EntityFramework.MicrosoftSqlServer": "7.0.0-*",
-    "Microsoft.Dnx.Compilation.CSharp.Abstractions": "1.0.0-*",
-    "xunit.assert": "2.1.0-*",
-    "xunit.extensibility.execution": "2.1.0-*"
+    "EntityFramework.Relational.Design.FunctionalTests": "7.0.0-*"
   },
   "compile": "..\\..\\src\\EntityFramework.Commands\\Design\\OperationHandlers.cs",
+  "testRunner": "xunit",
   "commands": {
     "test": "xunit.runner.aspnet"
```
Comment (natemcmaster) [inline test/EntityFramework.Commands.FunctionalTests/project.json]: Does this go away now?

File: test/EntityFramework.Microbenchmarks.Core/BenchmarkTestCaseRunner.cs [hunk 5]
```diff
@@ -128,12 +123,8 @@ private XunitTestRunner CreateRunner(int iteration, int totalIterations, string
 
         private static string GetFramework()
         {
-#if DNX451 || DNXCORE50
             var env = PlatformServices.Default.Runtime;
             return "DNX." + env.RuntimeType;
```
Comment (natemcmaster) [inline test/EntityFramework.Microbenchmarks.Core/BenchmarkTestCaseRunner.cs]: `"dotnet."` ? what is the new framework named?

**Comment by bricelam** [comment]
@divega Eventually `dnx451` will become `net451` and `dnxcore50` will become something like `netstandardapp10`, but we're still using the RC1 web tooling so things still need to be compatible with DNX in the short-term.

The concept of DNX has been removed from the product, and build.cmd will use the plain old Full .NET xUnit.net console runner. But in Visual Studio and for CoreCLR, we're still using the DNX runner. There will eventually be a .NET Core CLI runner, but not yet...

File: build.sh (hunk 1 above)
Comment (bricelam) [inline build.sh]: Soon-ish, I think NuGet will start distributing a self-contained, .NET Native-compiled binary for the various platforms.

File: src/EntityFramework.Core/project.json (hunk 3 above)
Comment (bricelam) [inline src/EntityFramework.Core/project.json]: In theory. Build scripts are selected independent of the referenced libraries.

File: test/EntityFramework.Commands.FunctionalTests/project.json (hunk 4 above)
Comment (bricelam) [inline test/EntityFramework.Commands.FunctionalTests/project.json]: No, still waiting for a .NET Core CLI runner and updated Web Tools.

File: test/EntityFramework.Microbenchmarks.Core/BenchmarkTestCaseRunner.cs (hunk 5 above)
Comment (bricelam) [inline test/EntityFramework.Microbenchmarks.Core/BenchmarkTestCaseRunner.cs]: Still TBD, but the current thinking is `netstandard` (.NET Platform Standard) and `netstandardapp` (.NET Platform Standard + console app model). And Full .NET will just be `net` (.NET Framework)

**Comment by natemcmaster** [comment]
LGTM. My comments weren't blocking concerns. :shipit: :see_no_evil:

**Comment by natemcmaster** [comment]
Is it going to an issue that [dorkbuild doesn't pass](https://github.com/aspnet/Universe/blob/dba3e55c0a4aa7ca59bc10852b3c35a000b1c615/KoreBuild-dotnet/build/_dotnet-test.shade#L47) `-parallel none` to `dotnet-test-xunit.exe`?

**Comment by bricelam** [comment]
@natemcmaster That code is [currently disabled](https://github.com/aspnet/Universe/blob/dev/KoreBuild-dotnet/build/_k-standard-goals.shade#L165).

File: src/EntityFramework.Commands/Design/OperationHandlers.cs [hunk 6]
```diff
@@ -1,7 +1,7 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-#if ENABLE_HANDLERS && !DNXCORE50
+#if ENABLE_HANDLERS && (NET451 || DNX451)
```
Comment (smitpatel) [inline src/EntityFramework.Commands/Design/OperationHandlers.cs]: Is `DNX451` defined/used here?

Comment (bricelam) [inline src/EntityFramework.Commands/Design/OperationHandlers.cs]: Yes, when the source file is included in the functional tests

**Pull request merged**